architecture/2024-08-resource-types-registration.md

```diff
@@ -0,0 +1,641 @@
+
+The rationale for this is to support registration/unregistration per-tenant of Radius rather than per-installation.
+
+#### Query Patterns
+
+The nested resource pattern is the best approach to support the data access patterns needed to implement UDTs. 
+
+Some examples:
+
+- UCP Request Routing:
+  - UCP knows the location, resource type, and API version.
+  - UCP can query a single `locations` resource to check for support and find the downstream API address.
+- Dynamic-RP Validation:
+  - Dynamic-RP knows the resource type and API version.
+  - Dynamic-RP can query a single `apiVersions` resource to fetch its schema for validation. 
+
+#### Resource Type Summaries
+
+The proposal also defines two new **non-resource** APIs:
+
+- `GET /planes/radius/{plane}/providers/`: list the most commonly used fields all resource providers.
+- `GET /planes/radius/{plane}/providers/{provider}` shows the most commonly used fields of specific resource provider.
+
+These APIs are optimized for queries and are consistent with similar APIs in ARM. To understand why this is important, remember that the OpenAPI schema for a resource type can be quite large, and the overall structure of the APIs is complex. Some functionality in Radius will need to query resource types frequently. For example, consider the need for the `rad recipe register` to validate a resource type. Building a separate API for queries allows us to optimize the behavior.
+
+These APIs can also be implemented consistently across all planes in the future. For example Azure doesn't support user-defined types, but it does support looking up the set of resource types.
+
+#### Locations
+
+UCP is a multi-regional control-plane like the control-plane that inspired it (ARM). We don't use UCP/Radius in any multi-region scenarios right now but it's important we avoid lock-in to a single-region design. ARM has a mature multi-location design but doesn't expose to users the ability to register types. However we can learn a lot from the scenarios supported by ARM for rolling out new API versions and new types.
+
+- Resource providers (non-declarative) are deployed per-location.
+- New functionality (API versions + types) is rolled out per-location.
+
+Some examples:
+
+> Resource provider `Contoso.Example` is only available in the `westus-1` location. UCP must reject creation requests to other locations.
+
+> Resource type `Contoso.Example/widgets` is available in all locations, but new API version `2024-08-01-preview` is only in the `eastus-2` location. UCP should block all requests for unsupported API version x location pairs.
+
+The design proposal includes axes for configuring location affinity:
+
+- `System.Resources/resourceProviders/locations` allows separate configuration of each location with an optional `address` in each location for the resource provider.
+- `System.Resources/resourceProviders/locations` includes the list of supported resource types and supported API versions.
```

Comment:
clarifying "includes the list of supported resource types and supported API versions".. for each location supported by a resource provider ?